### ISSUE TITLE
Semantics of "orless" and "ormore" definition

### ISSUE THREAD

**Issue by buzden** [saeaadl/emv2#13]
Semantics of `orless` and `ormore` operators should be defined.

We have some understanding of it. It is based on our intuition and is the following.

- `k ormore (ct1, ct2, …, ctn)` is equivalent to a disjunction of `M` _different_ disjuncts, where
 - `M` is a number of `k`-combinations of {ct1, ct2, …, ctn}-set and
 - each disjunct is a conjunction of `k`-combination of condition triggers from set {ct1, ct2, …, ctn} (i.e. length of each disjunct is `k` conjuncts).

- `k orless (ct1, ct2, …, ctn)` is equivalent to a disjunction of `M` _different_ disjuncts, where
 - `M` is a number of `(n - k)`-combinations of {ct1, ct2, …, ctn}-set and
 - each disjunct is a conjunction of `(n - k)`-combination of _negations_ of condition triggers from set {ct1, ct2, …, ctn} (i.e. length of each disjunct is `(n - k)` conjuncts).

Is this understanding correct? If it is not, could some definition be provided?

**Comment by reteprelief** [comment]
You are right the semantics are not clearly specified. (8) bullet 3 gives a partial specification for "ormore" in that it states in the case of 1 ormore the semantics are those of the non-exclusive or.
I think we want both an XOR and an OR operator. k ormore is then simply a shorthand for k-duples of OR.

Orless is a different story. At the last meeting we discussed two new operators "all" and "all but" (see #7).
Orless was intended to be an "all but" - however, you need to know the total number of elements. Originally, we do if the elements listed on the right of the operator are not multiplicities.
You could also use the noerror as a type value and effectively get a negation - but I wanted to avoid using negation.
I think we should replace the "orless" with the "all but".

**Comment by buzden** [comment]
Replacing of `orless` with `all but` looks like a great idea, but the semantics of `all but` still needs to be defined precisely. Our vision of it was described in the ticket https://github.com/saeaadl/emv2/issues/7#issuecomment-199717930 .

**Comment by jjhugues** [comment]
ormore and orless clarified in documented, issue marker "Issue #13"
